core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -336,43 +336,13 @@ protected boolean shouldCloseCacheStream() {
    */
   private void checkAndAdvanceBlockInStream() throws IOException {
     long currentBlockId = getCurrentBlockId();
-    if (mCurrentBlockInStream == null
-        || (mCurrentBlockInStream.remaining() == 0 && currentBlockId != mCurrentBlockId)) {
+    if (mCurrentBlockInStream == null || (mCurrentBlockInStream.remaining() == 0
+        && currentBlockId != mCurrentBlockId)) {
       closeCacheStream();
       updateBlockInStream(currentBlockId);
       mCurrentBlockId = currentBlockId;
       if (mShouldCacheCurrentBlock) {
-        try {
-          WorkerNetAddress address = mLocationPolicy.getWorkerForNextBlock(
-              mContext.getAlluxioBlockStore().getWorkerInfoList(), getBlockSizeAllocation(mPos));
-          // Don't cache the block to somewhere that already has it.
-          // TODO(andrew): Filter the workers provided to the location policy to not include
-          // workers which already contain the block. See ALLUXIO-1816.
-          if (mCurrentBlockInStream instanceof RemoteBlockInStream) {
-            WorkerNetAddress readAddress =
-                ((RemoteBlockInStream) mCurrentBlockInStream).getWorkerNetAddress();
-            // Try to avoid an RPC.
-            if (readAddress.equals(address)) {
-              mShouldCacheCurrentBlock = false;
-            } else {
-              BlockInfo blockInfo = mContext.getAlluxioBlockStore().getInfo(currentBlockId);
-              for (BlockLocation location : blockInfo.getLocations()) {
-                if (address.equals(location.getWorkerAddress())) {
-                  mShouldCacheCurrentBlock = false;
-                }
-              }
-            }
-          }
-          if (mShouldCacheCurrentBlock) {
-            mCurrentCacheStream = createCacheStream(currentBlockId, getBlockSize(mPos), address);
-          }
-        } catch (IOException e) {
-          logCacheStreamIOException(e);
-          mShouldCacheCurrentBlock = false;
-        } catch (AlluxioException e) {
-          LOG.warn(BLOCK_ID_NOT_CACHED, currentBlockId, e);
-          mShouldCacheCurrentBlock = false;
-        }
+        updateCacheStream();
```

Comment:
should this method just take in `currentBlockId` so it doesn't have to compute it again?